src/base/fs_unix.h

```diff
@@ -120,16 +122,23 @@ std::string get_app_path()
   std::vector<char> path(MAXPATHLEN);
 
 #if __APPLE__
-
   uint32_t size = path.size();
   while (_NSGetExecutablePath(&path[0], &size) == -1)
     path.resize(size);
+#elif __FreeBSD__
+  size_t size = path.size();
+  const int mib[] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+  while (sysctl(mib, 4, &path[0], &size, NULL, 0) == -1)
+      path.resize(size);
+#endif
 
-#else
-
+if (!path[0] && (access("/proc", F_OK) == 0)) {
```

Comment:
Is this required in OS X? why isn't this inside `__FreeBSD__`? Also this line needs a correct indentation.